pontoon/contributors/utils.py

```diff
@@ -427,24 +427,89 @@ def get_project_locale_contribution_counts(contributions_qs):
     return counts
 
 
-def get_contribution_timeline_data(user, contribution_type=None, day=None):
+def get_contribution_timeline_data(
+    user, year_shown=None, contribution_type=None, day=None
+):
     """
     Get data required to render the Contribution timeline on the Profile page
     """
     end = timezone.now()
-    start = end - relativedelta(months=1)
-    timeline_title = "Contribution activity in the last month"
+
+    if not year_shown:
+        start = end - relativedelta(day=1)
+        timeline_title = "Contribution activity in this month"
+    else:
+        timeline_title = "Contribution activity in the past year"
 
     if day is not None:
         start = datetime.datetime.fromtimestamp(day, tz=timezone.get_current_timezone())
         end = start + relativedelta(days=1)
         date = format_datetime(start, format="date")
         timeline_title = f"Contribution activity on { date }"
 
-    contribution_period = Q(created_at__gte=start, created_at__lte=end)
-    contributions_map = get_contributions_map(user, contribution_period)
+    contributions = {}
+
+    if year_shown:
+        # Perform query 12 times and append to contributions
+        for i in range(12):
+            month_start = end - relativedelta(months=i, day=1)
+            month_end = end - relativedelta(months=i, day=31)
+            contribution_period = Q(
+                created_at__gte=month_start, created_at__lte=month_end
+            )
+            month_start_name = month_start.strftime("%B %Y")
+            contributions_map = get_contributions_map(user, contribution_period)
```

Comment:
Let's not. `get_contributions_map` is already doing five separate ActionLog queries, so with this loop we get to 60 queries. All of which is then used only for numerical summaries.

All of this should really be one query, and we make the database do all the math. I could write the SQL for that, but making Django generate the same could be trickier. The next best alternative would be to query once for the actions, and then do the filtering in Python.